### PR TITLE
Fixed button behaviour and expanded on the GET api logic

### PR DESCRIPTION
**Pull request by Lukortech**
Merge before #14 if you want to go forwards :)

### CODE REVIEW

File: src/pages/Table/Table.tsx [hunk 1]
```diff
@@ -1,58 +1,64 @@
...
-      getapi('api/users/'+page+'/'+limit)
+  const initialDataState: DataI = {
+    users: [],
+    totalEntries: 0,
+  };
+  const [data, setData] = useState<DataI>(initialDataState);
+  const [params, setParams] = useState<ParamsI>({
+    page: 1,
+    limit: 10,
+  });
```
Comment (Lukortech) [inline src/pages/Table/Table.tsx]: Params are now a Observable variable, thanks to that we can tell React to make actions based on that change in Observable entries.

File: src/pages/Table/Table.tsx [hunk 2]
```diff
@@ -1,58 +1,64 @@
...
+  useEffect(() => {
+    async function getUsers(url: string, params?: ParamsI) {
+      const response = await fetch(url + params?.page + '/' + params?.limit);
+      const { users, totalEntries } = await response.json();
+
+      setData({ users: users.models, totalEntries: totalEntries });
+    }
+
+    getUsers('api/users/', params);
+  }, [params]);
```
Comment (Lukortech) [inline src/pages/Table/Table.tsx:41]: As you can see, we provide array of observables that our code is dependant on.
We observe for `params` and when they change the`getUsers` function is being called again with UPDATED values.

File: src/pages/Table/Table.tsx [hunk 3]
```diff
@@ -1,58 +1,64 @@
...
+
+  const handlePreviousPage = () => {
+    if (params.page > 0) {
+      setParams((prevState) => ({ ...prevState, page: prevState.page - 1 }));
```
Comment (Lukortech) [inline src/pages/Table/Table.tsx:45]: We take previous state of our params object and spread it, then overwrite page with new value based on the one we remember from just while ago (prevState.page)

File: src/pages/Table/Table.tsx [hunk 4]
```diff
@@ -1,58 +1,64 @@
...
     }
-  }
-  startFill();
+  };
+
+  const handleNextPage = () => {
+    if (params.page < Math.floor(data.totalEntries / params.limit)) {
+      setParams((prevState) => ({ ...prevState, page: prevState.page + 1 }));
+    }
```
Comment (Lukortech) [inline src/pages/Table/Table.tsx]: I am not too sure if this logic in if statement is correct. 
To be tested by @Qubaka

File: src/pages/Table/Table.tsx [hunk 5]
```diff
@@ -1,58 +1,64 @@
...
+  };
+
+  const handleLimitChange = (e: { target: { value: string } }) => {
+    setParams((prevState) => ({ ...prevState, limit: Number(e.target.value) }));
```
Comment (Lukortech) [inline src/pages/Table/Table.tsx]: This here allows us for api call whenever we change the limit of entries on page.

File: src/pages/Table/Table.tsx (hunk 4 above)
Comment (Qubaka) [reply src/pages/Table/Table.tsx]: Made some changes, now should work in every case

**Review by Sebucha — APPROVED** [review]

**Pull request merged**